### ISSUE TITLE
Change in python version of enviroment not reflected in kernel selection

### ISSUE THREAD

**Issue by grudloff** [microsoft/vscode-jupyter#6126]
## Environment data

-   VS Code version: 1.57.0
-   Jupyter Extension version : v2021.6.902307158
-   Python Extension version : v2021.6.897246044-dev
-   OS  and version: Windows_NT x64 10.0.19041
-   Python and Anaconda version: 3.8.10 and 4.10.1
-   Type of virtual environment used : conda
-   Jupyter server running: Local

If an environment is deleted, and a new one is created with the same name but the python version is different this information won't be updated on the kernel selection or the interpreter selection.

**Comment by claudiaregio** [comment]
@grudloff How are you deleting your environment? Does this still occur after you reload VS Code?

**Comment by grudloff** [comment]
I am deleting through the cli with `conda env remove --name env-name`. Yes, this persists after reopening VS Code.

**Comment by claudiaregio** [comment]
@grudloff If you go into your file directory and follow the path to the interpreter, is it still there? 

I created a test environment, confirmed I could use it and then removed it as you did. When I reloaded VS Code the option was still there. I followed the path below in my file directory and the env "test" was still there. I had to delete it and reload VS Code for the option to be removed. 

![image](https://user-images.githubusercontent.com/19276747/120912962-5d49e180-c648-11eb-853d-eed104d50325.png)

**Comment by DonJayamanne** [comment]
Duplicate of #6164

**Comment by DonJayamanne** [comment]
Sorry, not a duplicate of #6164

**Comment by DonJayamanne** [comment]
I can repro this, we store the display name & the like in the kernel spec.
We need a way to update this information.

**Comment by grudloff** [comment]
> @grudloff If you go into your file directory and follow the path to the interpreter, is it still there?
> 
> I created a test environment, confirmed I could use it and then removed it as you did. When I reloaded VS Code the option was still there. I followed the path below in my file directory and the env "test" was still there. I had to delete it and reload VS Code for the option to be removed.
> 
> ![image](https://user-images.githubusercontent.com/19276747/120912962-5d49e180-c648-11eb-853d-eed104d50325.png)

Yes! Conda doesn't delete the folder after the environment is removed.

**Comment by DonJayamanne** [comment]
Closing this issue, as this shouldn't be an issue anymore.
We don't display the kernels anymore, we display the intepreters in favor of the kernels.
